src/Product/SyncerHooks.php

```diff
@@ -315,24 +316,35 @@ protected function handle_delete_product( int $product_id ) {
 	}
 
 	/**
-	 * Schedules a job to send the product deletion notification
+	 * Maybe send the product deletion notification
 	 *
 	 * @since x.x.x
 	 * @param int $product_id
 	 */
 	protected function maybe_send_delete_notification( int $product_id ) {
 		$product = wc_get_product( $product_id );
-		if ( $product instanceof WC_Product && $this->product_helper->should_trigger_delete_notification( $product ) ) {
-			$this->product_helper->set_notification_status( $product, NotificationStatus::NOTIFICATION_PENDING_DELETE );
-			$this->product_notification_job->schedule(
-				[
-					'item_id' => $product->get_id(),
-					'topic'   => NotificationsService::TOPIC_PRODUCT_DELETED,
-				]
-			);
+
+		if ( $product instanceof WC_Product && $this->product_helper->has_notified_creation( $product ) ) {
+			$this->notifications_service->notify( NotificationsService::TOPIC_PRODUCT_DELETED, $product_id, [ 'offer_id' => $this->product_helper->get_offer_id( $product_id ) ] );
```

Comment:
For post deletions, it is better to send the notification directly.

Reply:
Why is it better to send the notification directly instead of scheduling the job?

Reply:
Because otherwise, it can be the case that the item is not in the database (for example, the user trash the product and deletes it from the trash before the action runs). In this cases an exception might happen or we cannot get anymore the product props.

Reply:
Thanks for the explanation. However, this could be problematic because if you delete multiple products or a variable product with many variations (say 30), we'll end up making all the requests simultaneously. From the merchant's perspective, it will take a while to delete a product with multiple variations because all the requests need to be completed. Also, we could reach the API rate limit.

As far as I can tell, we only need to notify the product ID, topic, and offer_id, so can't we just schedule the job to notify that? Or do we need the actual product?